src/components/runtime-common.js

```diff
@@ -108,6 +108,18 @@ export const PeriodicCookieSetter = () => {
   return null
 }
 
+export const PeriodicAzureCookieSetter = ({ proxyUrl }) => {
```

Comment:
I think it's worth extracting the commonality between `PeriodicCookieSetter` and `PeriodicAzureCookieSetter`. Something like:
```
export const CookieSetterWrapper = f => () => {
  const signal = useCancellation()
  usePollingEffect(withErrorIgnoring(f(signal)), { ms: 5 * 60 * 1000, leading: true })
  return null
}

export const PeriodicCookieSetter = CookieSetterWrapper(async signal => {
  await Ajax(signal).Runtimes.setCookie()
  cookieReadyStore.set(true)
})
```
Which could then also be used for `PeriodicAzureCookieSetter`.

Reply:
Yeah I thought of doing that... I chose not to though since its a small enough number of lines with enough params that the abstraction would be too complex to be worth it (i.e. they both use different cookie stores, call different APIs, and one takes a param and the other doesn't).